homeassistant/components/honeywell/__init__.py

```diff
@@ -1 +1,126 @@
 """Support for Honeywell (US) Total Connect Comfort climate systems."""
+from datetime import timedelta
+
+import somecomfort
+
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.util import Throttle
+
+from .const import _LOGGER, CONF_DEV_ID, CONF_LOC_ID, DOMAIN
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=180)
+PLATFORMS = ["climate"]
+
+
+async def async_setup_entry(hass, config):
+    """Set up the Honeywell thermostat."""
+    username = config.data[CONF_USERNAME]
+    password = config.data[CONF_PASSWORD]
+
+    client = await hass.async_add_executor_job(
+        get_somecomfort_client, username, password
+    )
+
+    if client is None:
+        return False
```

Comment:
```suggestion
        raise ConfigEntryNotReady
```

If we can retry, we should raise

Reply:
Looks like you only do this when auth fails.